src/ContractInteract/EIP20Gateway.js

```diff
@@ -246,6 +246,176 @@ class EIP20Gateway {
     return Promise.resolve(tx);
   }
 
+  /**
+   * Confirm redeem intent
+   *
+   * @param {string} redeemer Redeemer address.
+   * @param {string} nonce Redeemer nonce.
+   * @param {string} beneficiary Beneficiary address.
+   * @param {string} amount Redeem amount.
+   * @param {string} gasPrice Gas price that redeemer is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} blockNumber Block number.
+   * @param {string} storageProof Storage proof.
+   * @param {string} hashLock Hash lock.
```

Comment:
`storageProof` and `hashLock` should be in same order as in the function signature 🔍